lib/rubocop/cop/style/safe_navigation.rb

```diff
@@ -0,0 +1,104 @@
+# encoding: utf-8
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop transforms usages of a method call safeguarded by a call to
+      # `respond_to?` to safe navigation `&.`.
+      #
+      # @example
+      #   # bad
+      #   foo.bar if foo.respond_to?(:bar)
```

Comment:
I think that you are correct. I think I mixed up the functionality of for safe navigation as it pertains to a `nil` check vs a `respond_to?` check. I will need to update all of the functionality of this cop. This change should not be difficult to make. Good catch.